File: P1002 Find Common Characters/Solution.java
class Solution {
    // Time: O(n), where n is the total number of characters in A
    // Space: O(1)
    public List<String> commonChars(String[] A) {
        int[] count = new int[26];
        for(int i = 0; i < 26; i++) {
            count[i] = 100;
        }
        for(String s: A) {
            int[] indivialCount = new int[26]; 
            for(char c: s.toCharArray()) {
                indivialCount[c - 'a']++;
            }
            for(int i = 0; i < 26; i++) {
                count[i] = Math.min(indivialCount[i], count[i]);
            }
        }
        
        List<String> result = new ArrayList<String>();
        for(int i = 0; i < 26; i++) {
            for(int j = 0; j < count[i]; j++) {
                result.add(Character.toString((char) i + 'a'));
            }
        }
        
        return result;
    }
}

File: P1002 Find Common Characters/solution.py
class Solution:
    # Time: O(n), where n is the total number of characters in A
    # Space: O(1)
    def commonChars(self, A: List[str]) -> List[str]:
        count = 26 * [100]
        for s in A:
            individual_count = 26 * [0]
            for c in s:
                individual_count[ord(c) - ord('a')] += 1
            for i in range(26):
                count[i] = min(individual_count[i], count[i])
                
        result = []
        for i in range(26):
            for j in range(count[i]):
                result.append(chr(i + ord('a')))
        return result
